### ISSUE TITLE
Guide for multiple threads

### ISSUE THREAD

**Issue by forsakenzombi** [AHAAAAAAA/PokemonGo-Map#3177]
Guys I see people talking about multiple threads to have more scanners going at once but I for the life of me cannot find a guide that tells me how to set that up. I appologise if it's obvious but I sincerely cannot find it.

**Comment by lighter857** [comment]
use -t switch in the cmd line.. or modify in the config.ini.

Don't exceed 3 threads... so -t 3.

**Comment by aliamid93** [comment]
Yes, -t for more threads, but he meant have several different scanners across a place or city. You can run several instances of CMD with different locations ( add the -fl for fixed location for scanner) and have different ptc accounts for each running CMD instance. I tried it on my laptop, it worked however it was very slow.

**Comment by sodapuncher** [comment]
@aliamid93  Do you know how can I implement this on Jelastic? I'm not seeing an obvious spot to input commands. I had an idea to make a bat for it, but I'm a noob and don't know how... or how to use it, really.

**Comment by aliamid93** [comment]
@sodapuncher I have never used Jelastic, but just made an account and will try that there. If I figure it out, I will let you know how I did it.

**Comment by forsakenzombi** [comment]
Ok maybe i didnt understand what multiple threads actually meant. So from what I am gathering multiple threads just means that it will scan multiple circles at the same time. So if it was set to 3 then it would scan 3 circle areas around my location at a time right? 

If that is the case is there a way to move the location from the browser rather than stopping the command and reprogramming it? 

Additionally if I did want to you are saying that I could open multiple command prompts and say have 3 threads running in each as long as i use two different account to be able to scan two areas at a time?

Lastly is there any issues with leaving this running for extended periods of times? Could i say leave it scanning for hours and not really have to worry about it?

**Comment by forsakenzombi** [comment]
also when i do -t 2 or -t3 i see 2 or 3 respectively errors that say  "Search thread failed. Response dictionary key error" but it seems to search although when i move the point on the map to just a block away it stops finding pokemon.

**Comment by fucema** [comment]
Are you trying to run the master branch or develop branch? I run multiple accounts on the latest develop branch and it works fantastically. I can post instructions here if necessary.

**Comment by KSUWildcatFan** [comment]
@fucema Please do!

**Comment by forsakenzombi** [comment]
yes please do

**Comment by aliamid93** [comment]
@fucema Yes please show us!

**Comment by fucema** [comment]
$ git checkout develop

## Example 1: Run 5 PTC accounts that share the same password
$ python runserver.py -a ptc -u <username1> -u <username2> -u <username3> -u <username4> -u <username5> -p <common password> -st 10 -sd 5 -k <google map api key> -l "<initial scan location>" 

## Example 2: Run a mix of 3 Google and 2 PTC accounts that share the same password
$ python runserver.py -a google -u <username1@gmail.com> -a google -u <username2@gmail.com> -a google -u <username3@gmail.com> -a ptc -u <ptc username1> -a ptc -u <ptc username2> -p <common password> -st 10 -sd 5 -k <google map api key> -l "<initial scan location>"

Bam! Simple, right? Notice there is not "-t" threading option. The number of threads is determined by the number of accounts used -- 1 thread per account.

**Comment by clasherpekka** [comment]
So I have 10 accounts and so 10 threads. I run -st 11 and with my calculations I am scanning the same area every 2.758 minutes. Do I have less accuracy than someone using 5 accounts with an equally proportional steps?

Also, are beehive structures more accurate? Maybe equally as accurate.... If any data exist on that

**Comment by aliamid93** [comment]
@fucema what settings do you recommend to cover an area with the radius of 3km?

**Comment by aliamid93** [comment]
@clasherpekka how did you calculate that?

**Comment by forsakenzombi** [comment]
im using a windows machine. it doesnt recognize the git command. How would I change it from master branch?

**Comment by fucema** [comment]
@forsakenzombi 

You can download a zipped archive of the development branch from github.

**Comment by fucema** [comment]
@clasherpekka 
You may have better accuracy in the sense that you will be rescanning a previous step more frequently compared to a less parallelized scanner that has fewer threads. There is definitely a point of diminishing returns when it comes to the number of threads vs the step size. I use 7 account threads for a step size of 10 and it works well for me however.

Beehives are a different but related concept. Beehives are useful for fixed location scanning of very large areas. In the context of our cuirrent discussion, you would have multiple scanners participating in a beehive. Each scanner would have multiple account threads.

@aliamid93
I'm not sure to be honest, and I'm too lazy to calculate the number of steps (-st N) to cover 3km. But I bet you could figure it out with a little time and a calculator.
